src/vivarium_nih_us_cvd/components/healthcare_visits.py

```diff
@@ -79,20 +105,39 @@ def on_initialize_simulants(self, pop_data: SimulantData) -> None:
         index = pop_data.index
         event_time = self.clock() + self.step_size()
         df = self.population_view.subview(
-            [models.ISCHEMIC_STROKE_MODEL_NAME, models.MYOCARDIAL_INFARCTION_MODEL_NAME]
+            [
+                "age",
+                "sex",
+                models.ISCHEMIC_STROKE_MODEL_NAME,
+                models.MYOCARDIAL_INFARCTION_MODEL_NAME,
+            ]
         ).get(pop_data.index)
-        df[self.visit_type_column] = "none"
-        df[self.scheduled_visit_date_column] = pd.NaT
 
-        # Assign probabilities that each simulant will miss scheduled visits
+        # Initialize new columns
+        df[self.visit_type_column] = data_values.VISIT_TYPE.NONE
+        df[self.scheduled_visit_date_column] = pd.NaT
+        # FIXME [MIC-3457]: doc update changes missed appointment probability
         lower = data_values.MISS_SCHEDULED_VISIT_PROBABILITY_MIN
         upper = data_values.MISS_SCHEDULED_VISIT_PROBABILITY_MAX
         df[self.miss_scheduled_visit_probability_column] = lower + (
             upper - lower
         ) * self.randomness.get_draw(index)
+        df[self.sbp_medication_column] = np.nan
+        df[self.ldlc_medication_column] = np.nan
+        df[self.sbp_medication_adherence_type_column] = self.randomness.choice(
+            index,
+            choices=list(data_values.MEDICATION_ADHERENCE_TYPE_PROBABILITIY["sbp"].keys()),
+            p=list(data_values.MEDICATION_ADHERENCE_TYPE_PROBABILITIY["sbp"].values()),
+        )
+        df[self.ldlc_medication_adherence_type_column] = self.randomness.choice(
+            index,
+            choices=list(data_values.MEDICATION_ADHERENCE_TYPE_PROBABILITIY["ldlc"].keys()),
+            p=list(data_values.MEDICATION_ADHERENCE_TYPE_PROBABILITIY["ldlc"].values()),
+        )
 
-        # Handle simulants initialized in an post/chronic state
-        visitors = {}
+        df = self.initialize_medication_coverage(df)
```

Comment:
You have adopted a pattern here of generating a dataframe and then adding and removing columns from it. This makes it very hard to see what is actually changing in it when you go into and out of functions. You have created a very deep tree of function calls and it get very opaque as to what is happening in each one. Let's talk about this and see if there's a way to make this easier to read.

On a similar note, it's a good idea to have more descriptive variable names than this.

Reply:
This may be too late to be be worth mentioning, but it might have been simpler to have implemented this as if we had two separate decision trees - one for the treatement ramp and one for deciding whether or not to schedule a follow up.

`schedule_follow_up = already_on_treatment or sbp >=130`

I think this approach will allow you to eliminate a lot of the more confusing parts of the code.

Reply:
We discussed keeping the HospitalVisits class strictly for assigning visit types and scheduling followups and then creating two new treatment ramps (one for sbp and one for ldlc) to deal with treatments.